### ISSUE TITLE
Problem with sits_cube()

### ISSUE THREAD

**Issue by Drisse94** [e-sensing/sits#869]
Hello,

I'm trying to build a cube from ARD image collections with the following parameters :

CJBG_Cube <- sits_cube(
  source = "AWS",
  collection = "SENTINEL-S2-L2A-COGS",
  bands = c("B02", "B03", "B04", "B08", "CLOUD"),
  tiles = "31TGM",
  start_date = "2018-07-01",
  end_date = "2018-07-31"
)

When I run the statement, I get the following error message : "An error has occurred in a node and a recovery will be attempted at the end of the process / Error: .source_items_cube.stac_cube: empty cube metadata (could not retrieve cube metadata)"

I tried running the statement on another computer and it works fine. On the first computer (the one where the statement doesn't work), R version is 4.2.1 and RStudio version is 2022.7.1.554, while on the second computer R version is 4.2.1 and RStudio Version is 1.4.717. On both computers, sits version is 1.1.0. Could this be related to the RStudio version on the first computer?

Ideally, I would like the code to work on the first computer since I have to conduct other work on the second.

Thank you in advance for your reply

**Comment by OldLipe** [comment]
Hi @Drisse94. 

I believe the problem is not related with Rstudio version. This error occurs when no item is returned in the STAC API request. 
Sometimes, this error occurs when there is an issue on the service or an instability in your connection. 

Is this error still occurring on the first computer?

If you use the Microsoft Planetary Computer (MPC) does this error also occur? I was able to create the cubes using both providers.

```R
CJBG_Cube <- sits_cube(
    source = "MPC",
    collection = "SENTINEL-2-L2A",
    bands = c("B02", "B03", "B04", "B08", "CLOUD"),
    tiles = "31TGM",
    start_date = "2018-07-01",
    end_date = "2018-07-31"
)
```

**Comment by Drisse94** [comment]
Hi @OldLipe,

The error is still occuring on the first computer and when I try changing the source parameter to "MPC", RStudio aborts. Maybe it's related to the computer in itself? I ran the statement on the second computer again and it works fine.

Best
Daniel

**Comment by OldLipe** [comment]
@Drisse94, If you run it directly in R on the first computer, without using Rstudio, will the error still occur?

First time I have seen this behavior when creating a cube in sits.

**Comment by Drisse94** [comment]
Hi @OldLipe,

I tried running the code on R directly and it is still not working, but this time the error is different. It reads : "error reading from connection".

**Comment by OldLipe** [comment]
Hi @Drisse94,

Did you solve the issue? You could test the new version of sits to see if the problem still persists.

**Comment by Drisse94** [comment]
Hi Felipe,

It's Daniel, from Geneva. My apologies for answering so late, I contacted
you through my personal email which is linked to GitHub. Since I changed
computer I didn't follow up on this matter and since I finished my
internship I don't have access to the computer where sits_cube() didn't
work anymore. My bad, I know it's not good practice not to follow up on a
problem I've raised.

I submitted my paper on friday the 13th and am therefore finished with my
work with sits. I must stay it's in a great way thanks to you, I honestly
don't think I would have managed to go through the whole process without
your help. I know I've asked a lot of questions and I hope that was not to
much for you. You've been patient, reactive and educational, and I seize
the occasion to thank you for that. I mentioned you in my dissertation,
which I wrote in french, but I can still send it to you once I've defended
it on the 23rd of january, if you're interested.

I wish you a good weekend

Best regards
Daniel


Le mar. 17 janv. 2023 à 16:59, Felipe Carvalho ***@***.***> a
écrit :
